### ISSUE TITLE
Numerical instability in "quasistatic" calculations

### ISSUE THREAD

**Issue by gevero** [HomerReid/scuff-em#82]
Hi Homer

I am performing scattering and local field calculations with for the structure depicted below, using the scuff-scatter command line utility. All the files and inputs for the simulation are attacched below.

![bowtie3](https://cloud.githubusercontent.com/assets/7213490/13920732/da72b872-ef74-11e5-9ce0-ca3515d0ca2b.jpg)

As you can se in the plot reported below, I monitored the local field at the (0,0,z) coordinates, i.e. in the middle of the gap sweeping along the triangle thickness, for different wavelengths: 

- The first plot shows the results for the original structure, where the triangle thickness is 12nm the side is 120 nm, the gap is 3 nm and the angle curvature radius is 5 nm. Here everything works as aspected, and one can se the usual pronounced resonance at 800nm
- The second plot shows the results for a structure 100 times smaller but otherwise identical. I did this in order to perform a calculation in what should be a quasistatic approximation. As you can see I observe numerical problems, since the field mapping does not really makes sense. To name just one thing, it should be vertically symmetrical as in the first case, but it is not. 
Since during calculations I observe the message `WARNING: panel integration irregularities detected`, which never happens in the normal case. I believe that the problem comes from the mismatch between the size of the structure and the wavelength. I should i proceed to solve the problem? I know that there is a quasi-static command line utility, nevertheless I am not sure if it is fit for my purposes.

Best 

![spectral_cross_section](https://cloud.githubusercontent.com/assets/7213490/13920778/094f81ca-ef75-11e5-9eba-c52c62b07938.png)

[LambdaFile.in.txt](https://github.com/HomerReid/scuff-em/files/182302/LambdaFile.in.txt)
[input_bowtie3.args.txt](https://github.com/HomerReid/scuff-em/files/182303/input_bowtie3.args.txt)
[bowtie3.geo.txt](https://github.com/HomerReid/scuff-em/files/182306/bowtie3.geo.txt)
[bowtie3.EPfile.txt](https://github.com/HomerReid/scuff-em/files/182307/bowtie3.EPfile.txt)
[bowtie3.msh.txt](https://github.com/HomerReid/scuff-em/files/182308/bowtie3.msh.txt)
[e_ag_rashke_b_sc.edb.txt](https://github.com/HomerReid/scuff-em/files/182313/e_ag_rashke_b_sc.edb.txt)
[bowtie3.scuffgeo.txt](https://github.com/HomerReid/scuff-em/files/182314/bowtie3.scuffgeo.txt)

**Comment by gevero** [comment]
Hi homer

I tried to tackle the problem with a couple of different approaches:

-  Employ scuff-static: it failed miserably, could not say why, but I decided I would not spend too much time on it;
-  Instead of using small structures (a few, or a fraction of nanometer in size), I used micrometer size structure and remapped the optical constant and wavelength in the micrometer range;

The second approach seems to work brilliantly. Structure that gave highly unstable and unreliable results now converge robustly to physically meaningful results. Therefore I guess that the problem in the integration depends on the **absolute scale** of the modeled structure, and not on the **relative scale** with respect to the incident wavelength. If you can give me a nod of approval, I am closing the issue.

Thanks a lot

Giovanni

**Comment by HomerReid** [comment]
Yes, your solution was the right one. There is no intrinsic length scale in SCUFF-EM, so a mesh file describing a triangle of length `L=1` that is simulated at a wavelength of `Lambda=`1` could equally well describe a one-micron structure at 1-micron wavelength, or a 1-mm structure a 1-mm wavelength. The panel-integration algorithms work best when dimensionless numbers like `L/\Lambda` have numerical values within a few orders of magnitude of 1, i.e. in the range [0.001:1000].   

The only place where an intrinsic length scale is assumed is in evaluating frequency-dependent material properties, and the length scale assumed in this case is microns, i.e. the assumption is that `L=1` means that the object in question is 1 micron long. In API programs or python scripts this assumption can be changed by calling e.g. `MatProp::SetLengthUnit(1.0e-3)` (to set the default length unit to millimeters). For the command-line codes there is no direct functionality exposed to enable this (maybe there should be...) but the workaround you adopted, of modifying your description of the frequency dependence of the material, is fully equivalent and should work fine.

Regarding the failure of SCUFF-STATIC: Could it be simply that this involved the use of the wrong value of \epsilon? SCUFF-STATIC determines the DC permittivity of your structure by evaluating your  permittivity function at Omega=0. Your structure exhibits a sharp frequency dependence despite being in the quasistatic regime, which means what you are really observing is a sharp *permittivity* dependence: each calculation you do at a given wavelength is essentially a DC calculation with a different value of the DC permittivity. You should get equivalent results by doing SCUFF_STATIC calculations with the permittivity of your structure set to epsilon(omega) for each omega value you are using. If you find otherwise, let me know. (Or, if you don't want to bother since you already got things working your way, that's fine too.)
